blaze-server/src/main/scala/org/http4s/server/blaze/BlazeServer.scala

```diff
@@ -91,6 +104,7 @@ class BlazeBuilder(
         }
 
       case None =>
+        if (http2Support) logger.warn("Http2 support requires TLS.")
```

Comment:
That sounds fine for the case of fallback. Otherwise I think the best solution is an ```IllegalStateException``` when attempting to use http2 without TLS. I think I'd rather have the fallback, but I'm not strongly opinionated here.